### PR TITLE
aspectj 1.8.9 (new formula)

### PR DESCRIPTION
**Pull request by oldpatricka**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

Adds AspectJ, the AOP compiler for Java.

### CODE REVIEW

File: Formula/aspectj.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+class Aspectj < Formula
+  desc "AspectJ: aspect-oriented programming for Java"
+  homepage "https://eclipse.org/aspectj/"
+  url "http://www.eclipse.org/downloads/download.php?file=/tools/aspectj/aspectj-1.8.9.jar"
```
Comment (DomT4) [inline Formula/aspectj.rb]: You should be able to use `https://` okay here IIRC?

File: Formula/aspectj.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  sha256 "bab38539cbe6e932410a104c21e18f05b773c78001f8800993da29d718d31a1b"
+
+  bottle :unneeded
+  depends_on :java
```
Comment (DomT4) [inline Formula/aspectj.rb]: Please leave an empty line between the `bottle :` and the first `depends_on`, Thanks!

**Comment by oldpatricka** [comment]
Thanks @DomT4 those issues should be fixed. :)

File: Formula/aspectj.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    (bin/"ajbrowser").write <<-EOS.undent
+      #!/bin/sh
+      set -e
+      java -classpath "#{libexec}/aspectjtools.jar:$JAVA_HOME/lib/tools.jar:$CLASSPATH" -Xmx64M org.aspectj.tools.ajbrowser.Main "$@"
+    EOS
+    (bin/"ajdoc").write <<-EOS.undent
+      #!/bin/sh
+      set -e
+      java -classpath "#{libexec}/aspectjtools.jar:$JAVA_HOME/lib/tools.jar:$CLASSPATH" -Xmx64M org.aspectj.tools.ajdoc.Main "$@"
+    EOS
```
Comment (mikemcquaid) [inline Formula/aspectj.rb]: These scripts feel like quite a lot to be maintained just in the formula file. Could you see if the upstream project would be interested in accepting them?

Comment (apjanke) [inline Formula/aspectj.rb]: Yeah. Not just to reduce maintenance on the Homebrew side, but so all AspectJ users have consistent commands. Lack of full deployment support is a common annoyance with Java tools.

**Comment by oldpatricka** [comment]
@mikemcquaid @apjanke The problem is that the aspectj installer likes to hard code the installation directory with the scripts it writes out. That said, I agree bundling scripts isn't the best solution, especially if something gets added in the future, so I replaced that with some inreplace invocations.

**Comment by apjanke** [comment]
Okay, I think I see.

Our normal approach to Java stuff like this is to stuff the whole darn installation under libexec and then create wrapper scripts.

Hardcoding paths in scripts/links is normal; lots of packages do it for some things. Just need to make sure it points to the final destination. That `-to` option looks equivalent to `configure`'s `--prefix`. Maybe it should just be `-to #{libexec}/#{name}` instead of `-to .`, and then create thin wrapper scripts to account for Homebrew differences. There's a Homebrew method to create these; look around at other Java formulae.

**Comment by oldpatricka** [comment]
@apjanke Ah nice, that's a way cleaner approach! I did as you said and its way cleaner now. Thanks!

**Comment by mikemcquaid** [comment]
Yeh, this looks great now. Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**